plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -19,58 +19,73 @@
  * @return array{name: string, slug: string, short_description: string, requires: string|false, requires_php: string|false, requires_plugins: string[], download_link: string, version: string}|WP_Error Array of plugin data or WP_Error if failed.
  */
 function perflab_query_plugin_info( string $plugin_slug ) {
-	$plugin = get_transient( 'perflab_plugin_info_' . $plugin_slug );
-
-	if ( is_array( $plugin ) ) {
-		/**
-		 * Validated (mostly) plugin data.
-		 *
-		 * @var array{name: string, slug: string, short_description: string, requires: string|false, requires_php: string|false, requires_plugins: string[], download_link: string, version: string} $plugin
-		 */
-		return $plugin;
-	}
+	$transient_key = 'perflab_plugins_info';
+	$plugins       = get_transient( $transient_key );
 
-	$fields = array(
-		'name',
-		'slug',
-		'short_description',
-		'requires',
-		'requires_php',
-		'requires_plugins',
-		'download_link',
-		'version', // Needed by install_plugin_install_status().
-	);
+	if ( is_array( $plugins ) && isset( $plugins[ $plugin_slug ] ) ) {
```

Comment:
The second check here is problematic. We can't check that in the same assertions, otherwise a missing plugin will lead to endless re-issuing of the API request and bypassing the cache.

If the cache data is an array, we know it's a cache hit. So then we need to check _on_ that data if the `$plugin_slug` is not set, and if not return the same error as below.

Reply:
@felixarntz I have implemented this, could you please take a look.